api/tools/src/main/java/org/pmiops/workbench/tools/LoadDataDictionary.java

```diff
@@ -41,41 +43,48 @@ public CommandLineRunner run(
 
       for (Resource resource : getDataDictionaryExportFiles()) {
         InputStream is = resource.getInputStream();
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+        SimpleDateFormat df = new SimpleDateFormat("dd-MM-yyyy hh:mm:ss");
+        mapper.setDateFormat(df);
 
-        BufferedReader reader = new BufferedReader(new InputStreamReader(is));
-        Timestamp definedTime = new Timestamp(Long.parseLong(reader.readLine()) * 1000);
+        mapper.registerModule(new KotlinModule());
+        DataDictionary dd = mapper.readValue(is, DataDictionary.class);
 
-        String line;
-        while ((line = reader.readLine()) != null) {
-          String[] fields = line.split(":");
+        Timestamp newEntryDefinedTime = dd.getMeta_data()[0].getCreated_time();
 
-          String relevantOmopTable = fields[0];
-          String fieldName = fields[1];
+        CdrVersion cdrVersion = cdrVersionDao.findByName(dd.getMeta_data()[0].getCdr_version());
 
+        if (cdrVersion == null) {
+          // Skip over Data Dictionaries for CDR Versions not in the current environment
+          continue;
+        }
+
+        for (AvailableField field : dd.getTransformations()[0].getAvailable_fields()) {
           DataDictionaryEntry entry =
               dataDictionaryEntryDao.findByRelevantOmopTableAndFieldNameAndCdrVersion(
-                  relevantOmopTable, fieldName, defaultCdrVersion);
+                  field.getRelevant_omop_table(), field.getField_name(), cdrVersion);
 
           // We are skipping ahead if the defined times match by assuming that the definition has
           // not changed.
-          if (entry != null && definedTime.before(entry.getDefinedTime())) {
+          if (entry != null && newEntryDefinedTime.before(entry.getDefinedTime())) {
             continue;
           }
 
           if (entry == null) {
             entry = new DataDictionaryEntry();
-            entry.setRelevantOmopTable(relevantOmopTable);
-            entry.setFieldName(fieldName);
-            entry.setCdrVersion((defaultCdrVersion));
+            entry.setRelevantOmopTable(field.getRelevant_omop_table());
+            entry.setFieldName(field.getField_name());
+            entry.setCdrVersion(defaultCdrVersion);
           }
 
-          entry.setDefinedTime(definedTime);
-          entry.setOmopCdmStandardOrCustomField(fields[2]);
-          entry.setDescription(fields[3]);
-          entry.setFieldType(fields[4]);
-          entry.setDataProvenance(fields[5]);
-          entry.setSourcePpiModule(fields[6]);
-          entry.setTransformedByRegisteredTierPrivacyMethods("true".equals(fields[7]));
+          entry.setDefinedTime(newEntryDefinedTime);
+          entry.setOmopCdmStandardOrCustomField(field.getOmop_cdm_standard_or_custom_field());
```

Comment:
Can your `ObjectMapper` help with some of this boilerplate?

Reply:
I looked into it but I don't think it'll save us any code in this case because of the name mismatches. It is possible to override name mismatches but we would then be looking at at least an equal number of lines from override statements.

Reply:
Yeah, but those lines would live in the mapper where it can be reused and not spilled out in this unrelated function, so I think it's still a win.